kroxylicious-filters/kroxylicious-authorization/src/main/java/io/kroxylicious/filter/authorization/OffsetFetchUpToV7Enforcement.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.filter.authorization;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+
+import org.apache.kafka.common.message.OffsetFetchRequestData;
+import org.apache.kafka.common.message.OffsetFetchResponseData;
+import org.apache.kafka.common.message.RequestHeaderData;
+import org.apache.kafka.common.protocol.Errors;
+import org.apache.kafka.common.record.RecordBatch;
+import org.apache.kafka.common.requests.OffsetFetchResponse;
+
+import io.kroxylicious.authorizer.service.Action;
+import io.kroxylicious.authorizer.service.AuthorizeResult;
+import io.kroxylicious.authorizer.service.Decision;
+import io.kroxylicious.proxy.filter.FilterContext;
+import io.kroxylicious.proxy.filter.RequestFilterResult;
+
+public class OffsetFetchUpToV7Enforcement extends ApiEnforcement<OffsetFetchRequestData, OffsetFetchResponseData> {
```

Comment:
I wonder if we really should have the version number in the class name at all. Really the difference is that the one case supports batching and the other doesn't. `OffsetFetchBatchingEnforcement` and `OffsetFetchNonBatchingEnforcement` makes it a bit clearer from the outside what's going on.